dev/io.openliberty.jakarta.concurrency.3.1_fat_tck/fat/src/io/openliberty/jakarta/enterprise/concurrent/tck/FATSuite.java

```diff
@@ -25,4 +30,39 @@
                 ConcurrentTckLauncherWeb.class //LITE MODE
 })
 public class FATSuite {
+
+    public static boolean shouldRunSignatureTests(Class<?> testClass) {
+        boolean result = false;
+        String reason = "";
+
+        try {
+            if (!testClass.isAnnotationPresent(Mode.class)) {
+                reason = testClass.getCanonicalName() + " is not run in full mode.";
+                return result = false;
+            }
+
+            if (testClass.getAnnotation(Mode.class).value() != TestMode.FULL) {
+                reason = testClass.getCanonicalName() + " is not run in full mode.";
+                return result = false;
+            }
+
+            if (System.getProperty("os.name").contains("Windows")) {
+                reason = "signature test plugin not supported on Windows.";
+                return result = false;
+            }
+
+            if (JavaInfo.JAVA_VERSION != 17 || JavaInfo.JAVA_VERSION != 21) {
```

Comment:
Would it be useful to add "&& JavaInfo.JAVA_VERSION < 26".  The current code would just silently never run the tests after Java 21.  Perhaps that is ok, or we could force someone to reconsider after say Java 25?  Would we never plan to certify Jakarta EE 11 on any LTS after 21?

Reply:
Good suggestion, but the Concurrency TCK has generated signature files for specifically Java 17 & Java 21 which is why this logic is so strict.